x/valset-pref/msg_server_test.go

```diff
@@ -10,16 +10,18 @@ func (suite *KeeperTestSuite) TestSetValidatorSetPreference() {
 	suite.SetupTest()
 
 	// setup 3 validators
-	valAddrs := suite.SetupMultipleValidators(3)
+	valAddrs := suite.SetupMultipleValidators(6)
 
 	tests := []struct {
-		name        string
-		delegator   sdk.AccAddress
-		preferences []types.ValidatorPreference
-		expectPass  bool
+		name                   string
+		delegator              sdk.AccAddress
+		preferences            []types.ValidatorPreference
```

Comment:
What's the significance of having the preference field and testing different preferences in this test case? From my understanding this test case seems like it's testing setting val set preference under conditions of existing delegations / having existing val set

Reply:
hmm so with the preference we can test create/update valset and also check convert existing delegation to valset delegations. Also all validators and weight on this field needs to valid so it's checking all of it. Not sure if that fully answers your question but happy to discuss more :))